resolwe/flow/tests/test_executors.py

```diff
@@ -359,3 +359,107 @@ def test_executor_fs_lock(self):
         self.assertEqual(data.process_error, [])
         # Check that temporary file was not deleted.
         self.assertTrue(file.exists())
+
+    @with_docker_executor
+    @tag_process("test-save-number")
+    @override_settings(FLOW_MANAGER_KEEP_DATA=True)
+    def test_executor_keep_data_true(self):
+        # Run the process for the first time.
+        data = self.run_process("test-save-number", {"number": 42})
+
+        # Make sure that process was successfully ran first time.
+        self.assertEqual(data.output["number"], 42)
+        data.output = {}
+        data.save()
+
+        json_file = Path(data.location.get_path(filename="jsonout.txt"))
+        stdout_file = Path(data.location.get_path(filename="stdout.txt"))
+
+        self.assertTrue(json_file.exists())
+        self.assertTrue(stdout_file.exists())
+
+        # Remove stdout and json files to run the process again.
+        json_file.unlink()
+        stdout_file.unlink()
+
+        # Create the temporary file that should NOT be purged since keep data
+        # is set to True.
+        tmp_file = Path(
+            data.location.get_path(filename="temporary_file_do_not_purge.txt")
+        )
+        self.assertFalse(tmp_file.exists())
+        tmp_file.touch()
+
+        # Run the process again.
+        process = subprocess.run(
+            ["python", "-m", "executors", ".docker"],
+            cwd=data.get_runtime_path(),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            timeout=5,
+        )
+
+        # Drain control events generated by this executor.
+        manager.reset(keep_state=True)
+        self.assertEqual(process.returncode, 0)
+
+        # Check the status of the data object.
+        data.refresh_from_db()
+        # Make sure the process completed successfully.
+        self.assertEqual(data.output["number"], 42)
+        self.assertEqual(data.status, Data.STATUS_DONE)
+        self.assertEqual(data.process_error, [])
+        # Make sure that the temporary file was not deleted.
+        self.assertTrue(tmp_file.exists())
+
+    @with_docker_executor
+    @tag_process("test-save-number")
+    @override_settings(FLOW_MANAGER_KEEP_DATA=False)
+    def test_executor_keep_data_false(self):
```

Comment:
I think that this test is not necessary as this is the default behavior tested in the "purge" tests.